ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/results_manager.py

```
@@ -0,0 +1,144 @@
+        """Initialize the ResultsManager with default values and stats counters set to 0.
+
+        Args:
+        ----
+            only_failed_tests (bool): Flag to determine if only failed tests are saved.
+                                      Defaults to True. If set to False, all tests will be saved.
+        """
+        self.test_id = 0
+        self.total_tests_passed = 0
+        self.total_tests_failed = 0
+        self.total_tests_skipped = 0
+        self.total_tests_not_run = 0
+        self.dut_stats = defaultdict(
+            lambda: {
+                "tests_passed": 0,
+                "tests_failed": 0,
+                "tests_skipped": 0,
+                "tests_not_run": 0,
+                "categories_failed": set(),
+                "categories_skipped": set(),
+            },
+        )
+        self.category_stats = defaultdict(
+            lambda: {
+                "tests_passed": 0,
+                "tests_failed": 0,
+                "tests_skipped": 0,
+                "tests_not_run": 0,
+            },
+        )
+        self.failed_tests = []
+        self.all_tests = []
+        self.only_failed_tests = only_failed_tests
+
+    def _parse_result(self, result: dict) -> dict:
+        """Parse a single test result and converts it into a standardized format for the reports.
+
+        Args:
+        ----
+            result (dict): The test result data to parse.
+
+        Returns:
+        -------
+            dict: The parsed and standardized test result.
+        """
+        test_categories = [
+            " ".join(word.upper() if word.lower() in ACRONYM_CATEGORIES else word.title() for word in category.split())
+            for category in result.get("categories", [])
+        ]
+
+        # Mapping the ANTA results
+        result_mapping = {"success": "PASS", "failure": "FAIL", "error": "FAIL", "skipped": "SKIPPED", "unset": "NOT RUN"}
+        anta_result = result.get("result", "")
```

should we not log something here if we don't get a `result` ?
Same if the `anta_result` does not exist in the mapping

we are silently ignoring stuff

---

This is coming from ANTA `TestResult` which is a Pydantic model. We technically shouldn't be missing these fields.

---

If it's missing, it will appear in the final report. Do you think we should still log it?